src/scheduler/resource.cpp

```diff
@@ -696,71 +340,66 @@ create_resource_signature(schd_resource *reslist, std::unordered_set<resdef *>&
 
 
 /**
- * @brief
- * 		query the resource definition from the server and create derived
- *        data structures.  Only query if required.
+ * @brief update allres and sub-containers of resource definitions.  This is called
+ *		in schedule().  If it fails in schedule() we'll pick it up in the next call to quuery_server()
  *
  * @param[in]	pbs_sd	-	connection descriptor to the pbs server
  *
- * @return	int
- * @retval	1	: success - we've updated the global resdef arrays
- * @retval	0	: failure - we haven't updated the global resdef arrays.  Scheduling
- *                     		should not continue.
+ * @return	bool
+ * @retval	true - successfully updated resdefs
+ * @retval	false - failed to update resdefs
  */
-int
+bool
 update_resource_defs(int pbs_sd)
 {
-	int error = 0;
-	/* only query when needed*/
-	if (allres != NULL)
-		return 1;
-
-	allres = query_resources(pbs_sd);
-
-	if (allres != NULL) {
-		consres = (resdef**) filter_array((void **) allres,
-			def_is_consumable, NULL, NO_FLAGS);
-		if (consres == NULL)
-			error = 1;
-
-		if (!error) {
-			boolres = (resdef**) filter_array((void **) allres,
-				def_is_bool, NULL, NO_FLAGS);
-			if (boolres == NULL)
-				error = 1;
-		}
-
-		if (!conf.res_to_check.empty()) {
-			conf.resdef_to_check = resstr_to_resdef(conf.res_to_check);
-			if (conf.resdef_to_check.empty())
-				error = 1;
+	clear_limres();
```

Comment:
Should we move this to the end of the function? In this change, We might end up clearing it when when query_resources fail.